cmake/UseCcache.cmake

```diff
@@ -0,0 +1,56 @@
+# Compiling ccache via ccache is fine, because this uses a stable version which
+# is installed on the system.
+
+# Modified version of Craig Scott's "Professional CMake: A Practical Guide", 7th Edition
+function(use_ccache)
+    if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+        message(INFO " use_ccache() disabled, as not called at root of project")
+        return()
+    endif()
+
+    find_program(CCACHE_PROGRAM ccache)
+    if(NOT CCACHE_PROGRAM)
+        message(INFO " use_ccache() disabled, as ccache program not found")
```

Comment:
Are the `message(INFO " ...")` calls temporary debug printouts that should be removed or should they be kept? If the latter I think that they should be more succinctly phrased and use `message(STATUS "...")` instead.

Reply:
Actually cmake does not have INFO according to documentation. This prints NOTICE with text "INFO use_ccache() disabled, as ccache program not found".
NOTICE is more "significant" (or something) than STATUS.
So I'll change this to STATUS.

https://cmake.org/cmake/help/latest/command/message.html